### DISCUSSION TITLE
Plotting the true values of a binary array

### DISCUSSION THREAD

**Discussion by koenterheegde0507** [pyvista/pyvista#5778] [Q&A]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import os
import tkinter as tk
from tkinter import filedialog
import pyvista as pv
import numpy as np
import skimage
from skimage.morphology import skeletonize_3d
import matplotlib.pyplot as plt
import skeletor as sk
from stl import mesh
import trimesh

root = tk.Tk()
root.withdraw()
root.title('Pick patient data')
file_path = filedialog.askopenfilename()

chosenMesh = pv.read(f'{file_path}')
density = chosenMesh.length/200
voxels = pv.voxelize(chosenMesh, density)

voxel_plotter = pv.Plotter()
voxel_plotter.add_mesh(voxels)
voxel_plotter.show()

x_min, x_max, y_min, y_max, z_min, z_max = chosenMesh.bounds
x = np.arange(x_min, x_max, density)
y = np.arange(y_min, y_max, density)
z = np.arange(z_min, z_max, density)
x, y, z = np.meshgrid(x, y, z)

grid = pv.StructuredGrid(x, y, z)
ugrid = pv.UnstructuredGrid(grid)

selection = ugrid.select_enclosed_points(chosenMesh.extract_surface(),
tolerance=0.0,
check_surface=False)

mask = selection.point_data['SelectedPoints'].view(bool)
mask = mask.reshape(x.shape)

custom_colormap = ['purple','red']
pv.plot(grid.points, scalars=mask.astype(int), cmap=custom_colormap)
```

### Description

I have an issue that is related to issue #4339 . I want to plot the 'True' points of a binary array in 3D, starting from a voxelized .stl file. I have added my code and screenshots, and my question is why the last plot is not looking like the surface of the .STL but then made from points.  
Voxelized mesh: 
<img width="960" alt="pyvista_voxelization" src="https://github.com/pyvista/pyvista/assets/152174547/20c936a9-4363-4f8b-be3d-8e07c41c2acc">

Grid with points showing true/false (this is correct, if you zoom in you can see the red dots that make up the voxelized mesh): 
<img width="960" alt="pyvista_binaryImage" src="https://github.com/pyvista/pyvista/assets/152174547/09c56322-f34c-4c8a-a99d-539110de2bc2">

I want to have a plot that now only shows the red points from the image above (not by making the False points white). However, i get this result:
![image](https://github.com/pyvista/pyvista/assets/152174547/cc606005-5ef0-41da-8d0a-e785191784bf)

### System Information

```shell
OS : Windows
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 7.8 GiB
       Environment : Python
        GPU Vendor : Intel
      GPU Renderer : Intel(R) UHD Graphics 620
       GPU Version : 4.5.0 - Build 30.0.101.1692
  MathText Support : False

  Python 3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit
  (AMD64)]

           pyvista : 0.42.3
               vtk : 9.3.0
             numpy : 1.24.3
        matplotlib : 3.8.2
            scooby : 0.9.2
             pooch : 1.8.0
            pillow : 10.1.0
           imageio : 2.33.0
             PyQt5 : 5.15.4
           IPython : 8.13.2
        ipywidgets : 8.1.2
             scipy : 1.9.1
              tqdm : 4.66.2
            meshio : 5.3.5
      nest_asyncio : 1.5.6
```

**Comment by akaszynski** [comment]
You'll want to extract the points that are `True` within your `mask` by either using [extract_points](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.DataSetFilters.extract_points.html#pyvista.DataSetFilters.extract_points) with:
```py
extracted_grid_points = grid.extract_points(mask)
```

 by simply creating a new array of extracted points with:

```py
extracted_points = grid.points[mask]
```

**Reply by koenterheegde0507** [reply]
`extracted_grid_points = grid.extract_points(mask)
extracted_points = grid.points[mask]

pv.plot(extracted_grid_points)
pv.plot(extracted_points)`

`pv.plot(extracted_points)` returns the following figure:
![image](https://github.com/pyvista/pyvista/assets/152174547/e2021da5-d448-40e3-a48e-577588144914)

Using `extracted_points = grid.points[mask]` returns the following error message: `IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed`

**Reply by akaszynski** [reply]
Please zip your dataset if possible and I'll take a quick look at it.

**Reply by koenterheegde0507** [reply]
Sorry for the super late reply! Unfortunately I can't share the zip due to the size being too big.

**Reply by akaszynski** [reply]
Email me at akascap@gmail.com if you still need help with this.
